### PR TITLE
feat: enable Rest api using grpc-gateway

### PR DESCRIPTION
**Pull request by sbadla1**
Closes # 

## 📑 Description

enable Rest api using grpc-gateway. This is related to PR https://github.com/k8sgpt-ai/schemas/pull/20

## ✅ Checks

- [ ] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [ ] All the tests have passed

### CODE REVIEW

File: cmd/serve/serve.go [hunk 1]
```diff
@@ -153,4 +155,5 @@ func init() {
 	ServeCmd.Flags().StringVarP(&port, "port", "p", "8080", "Port to run the server on")
 	ServeCmd.Flags().StringVarP(&metricsPort, "metrics-port", "", "8081", "Port to run the metrics-server on")
 	ServeCmd.Flags().StringVarP(&backend, "backend", "b", "openai", "Backend AI provider")
+	ServeCmd.Flags().BoolVarP(&enableRest, "enable-REST", "r", false, "Enable REST api using gppc-gateway")
```
Comment (AlexsJones) [inline cmd/serve/serve.go:158]: Perhaps we can use `--http` flag to enable rest API?

File: cmd/serve/serve.go [hunk 2]
```diff
@@ -33,6 +33,7 @@ var (
 	port        string
 	metricsPort string
 	backend     string
+	enableRest  bool
```
Comment (thschue) [inline cmd/serve/serve.go:36]: please change this to enableHttp (or similar)

File: cmd/serve/serve.go (hunk 1 above)
Comment (thschue) [inline cmd/serve/serve.go:158]:
```suggestion
	ServeCmd.Flags().BoolVarP(&enableHttp, "http", "h", false, "Enable REST/HTTP api using grpc-gateway")
```

File: cmd/serve/serve.go (hunk 2 above)
Comment (thschue) [inline cmd/serve/serve.go:36]:
```suggestion
	enableHttp  bool
```

File: cmd/serve/serve.go [hunk 3]
```diff
@@ -126,6 +127,7 @@ var ServeCmd = &cobra.Command{
 			Backend:     aiProvider.Name,
 			Port:        port,
 			MetricsPort: metricsPort,
+			EnableRest:  enableRest,
```
Comment (thschue) [inline cmd/serve/serve.go:130]:
```suggestion
			EnableRest:  enableHttp,
```

**Review by thschue — CHANGES_REQUESTED** [review]
I like the possibility to have a rest/http-api, made some suggestions for naming and I think you need to rebase

Thank you for this contribution!

File: cmd/serve/serve.go (hunk 1 above)
Comment (sbadla1) [reply cmd/serve/serve.go]: updated

File: cmd/serve/serve.go (hunk 2 above)
Comment (sbadla1) [reply cmd/serve/serve.go]: updated

File: cmd/serve/serve.go (hunk 1 above)
Comment (sbadla1) [reply cmd/serve/serve.go]: updated

File: cmd/serve/serve.go (hunk 2 above)
Comment (sbadla1) [reply cmd/serve/serve.go]: updated

File: cmd/serve/serve.go (hunk 3 above)
Comment (sbadla1) [reply cmd/serve/serve.go]: updated

**Comment by sbadla1** [comment]
@AlexsJones @thschue Thanks for the review. Please note that my changes need to reference methods([example](https://github.com/k8sgpt-ai/k8sgpt/pull/834/files#diff-104a11712684e520f0affa95e6a053faeb713306a10f69ee99b9729cc9dd1996R103)) generated by grpc-gateway. Grpc-gateway changes and code generation part is in this [PR](https://github.com/k8sgpt-ai/schemas/pull/20)

**Comment by AlexsJones** [comment]
Sorry for this @sbadla1 going to get this sorted out today, updating our schemas

**Pull request merged**